assets/js/admin.js

```diff
@@ -1000,6 +1000,29 @@
       });
     }
 
+     //   Maximum number of related posts / Maximum related posts to display
+     const inputMax = $('#input_max');
+     const numberDisplay = $('#number');
+     const helpText = $('#taxopress_maxposts_helptext');
+ 
+     function validateInput() {
+         const maxVal = parseInt(inputMax.val(), 10); //used 10 to specify the decimal
+         const displayVal = parseInt(numberDisplay.val(), 10);
+ 
+         if (displayVal > maxVal) {
+             helpText.show();
+             numberDisplay[0].setCustomValidity('This value cannot exceed the maximum number of related posts.');
+         } else {
+             helpText.hide();
+             numberDisplay[0].setCustomValidity('');
+         }
+     }
+ 
+     // Attach event listeners to both fields to validate on change
+     $('#input_max, #number').on('input', validateInput);
```

Comment:
Using #number selector for example is too generic and could affect other pages that may have same ID.

So, check the page elements and add it parent selector. For example, .parent-class-name #number etc. @Deji98

Reply:
done that @ojopaul